app/workers/expired_downtime_cleaner.rb

```diff
@@ -0,0 +1,21 @@
+class ExpiredDowntimeCleaner
+  include Sidekiq::Worker
+
+  def self.enqueue(downtime)
+    dequeue_existing_jobs(downtime)
+    perform_at(downtime.end_time, downtime.id.to_s)
```

Comment:
Is it worth adding a small margin here?  I'm not sure how precise sidekiq is with its scheduling, but we don't want to end up with a race between this time and the `downtime.end_time.future?` call below.  Adding something like 30 seconds here would guarantee to avoid that.